tests/llm/test_example_selector.py

```diff
@@ -0,0 +1,218 @@
+import asyncio
+from typing import Any, Dict, List
+
+import pytest
+import numpy as np
+from langchain_core.example_selectors.base import BaseExampleSelector
+from langchain_core.example_selectors import LengthBasedExampleSelector
+from langchain_core.prompts import FewShotPromptTemplate, PromptTemplate
+from langchain_core.messages import AIMessage, HumanMessage
+from pydantic import BaseModel, Field, RootModel
+
+from chatsky import MISC, RESPONSE
+from chatsky.llm.example_selector import Example, StaticExampleSelector, to_langchain_context
+from chatsky.responses.llm import LLMResponse
+
+np.random.seed(0)
+
+
+class ExamplePrompt(BaseModel, arbitrary_types_allowed=True):
+    examples: StaticExampleSelector | BaseExampleSelector
+
+    async def __call__(self):
+        examples: list[dict[str, str]] = await self.examples.select_examples({"input": ctx.last_request.text})
+
+
+class ResponseModel(BaseModel):
+    sum: float = Field(description="Sum of the numbers")
+    prod: float = Field(description="Product of the numbers")
+
+
+class RequestModel(BaseModel):
+    operand_1: float = Field(description="The first operand")
+    operand_2: float = Field(description="The second operand")
+
+
+class MockCustomExampleSelector(BaseExampleSelector, RootModel):
+
+    root: List[Example]
+
+    def add_example(self, example: Dict[str, str]) -> Any:
+        self.root.append(Example.model_validate(example))
+
+    def unpack_model(self, example_part: str | BaseModel) -> str:
+        return str(example_part.model_dump_json()) if isinstance(example_part, BaseModel) else example_part
+
+    def select_examples(self, input_variables: Dict[str, str]) -> List[dict]:
+        subset = np.random.choice(self.root, input_variables["size"], input_variables["replace"])
+        return [
+            {"input": self.unpack_model(example.input), "output": self.unpack_model(example.output)}
+            for example in subset
+        ]
+
+
+### Tests
+
+
+@pytest.fixture()
+def static_selector_fixture(scope="function"):
```

Comment:
Scope is specified in the `fixture` call.
https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session